### PR TITLE
builtins: support length, octet_length and bit_length for bit types

### LINKED ISSUE #45847: sql: length of bit type is not supported
Repro:
```
root@:26257/defaultdb> SELECT length('0110'::bit(4));
ERROR: unknown signature: length(bit)
```
EXPECTED: No error, PG gives this:
```
template1=# SELECT length('0110'::bit(4));
 length
--------
      4
```

### PR DESCRIPTION
**Pull request by otan**
Resolves #45847.

Release justification: low risk, high benefit changes to existing
functionality

Release note (sql change): Add the ability to use the length,
octet_length and bit_length builtins on BIT and VARBIT.

### CODE REVIEW

File: pkg/sql/sem/tree/testdata/eval/builtins [hunk 1]
```diff
@@ -82,3 +82,23 @@ eval
...
+----
+4
+
+eval
+bit_length('1011'::bit(4))
+----
+4
+
+eval
+octet_length('1011'::bit(4))
```
Comment (rohany) [inline pkg/sql/sem/tree/testdata/eval/builtins]: sorry i'm confused why this returns something different than below

Comment (otan) [reply pkg/sql/sem/tree/testdata/eval/builtins]: this one is 4 bits, which needs 1 byte.
the below is 12 bits, which needs 2 bytes.

Comment (rohany) [reply pkg/sql/sem/tree/testdata/eval/builtins]: but its the same bit string idgi

Comment (otan) [reply pkg/sql/sem/tree/testdata/eval/builtins]: a bit string is represented as bytes in memory

Comment (rohany) [reply pkg/sql/sem/tree/testdata/eval/builtins]: oh i was confused by the size parameter in `BIT`

**Comment by rohany** [comment]
lgtm

**Comment by otan** [comment]
thanks!

bors r=rohany

**Pull request merged**